autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java

```diff
@@ -81,35 +85,47 @@
     @Inject
     private UsageReportingService usageReportingService;
 
-    public ScalingRequest(Cluster cluster, ScalingPolicy policy, int totalNodes, int desiredNodeCount, List<String> decommissionNodeIds) {
+    @Inject
+    private LimitsConfigurationService limitsConfigurationService;
+
+    public ScalingRequest(Cluster cluster, ScalingPolicy policy, int existingClusterNodeCount, int existingHostGroupNodeCount,
+            int desiredHostGroupNodeCount, List<String> decommissionNodeIds) {
         this.cluster = cluster;
         this.policy = policy;
-        this.totalNodes = totalNodes;
-        this.desiredNodeCount = desiredNodeCount;
+        this.existingClusterNodeCount = existingClusterNodeCount;
+        this.existingHostGroupNodeCount = existingHostGroupNodeCount;
+        this.desiredHostGroupNodeCount = desiredHostGroupNodeCount;
         this.decommissionNodeIds = decommissionNodeIds;
     }
 
     @Override
     public void run() {
         LoggingUtils.buildMdcContext(cluster);
         try {
-            int scalingAdjustment = desiredNodeCount - totalNodes;
+            int scalingAdjustment = desiredHostGroupNodeCount - existingHostGroupNodeCount;
             if (!decommissionNodeIds.isEmpty()) {
                 scaleDownByNodeIds(decommissionNodeIds);
             } else if (scalingAdjustment > 0) {
-                scaleUp(scalingAdjustment, totalNodes);
+                int cbSupportedMaxClusterSize = limitsConfigurationService.getMaxNodeCountLimit();
+                if ((existingClusterNodeCount + scalingAdjustment) > cbSupportedMaxClusterSize) {
+                    LOGGER.info("Upscale requested for '{}' nodes. Cluster can have maximium '{}' node(s)", scalingAdjustment, cbSupportedMaxClusterSize);
```

Comment:
Would be useful to log current node count (ideally with hostGroup break-up - which, am not sure is available)

Reply:
logged the current node count.